File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.py
class Solution(object):
    def countConsistentStrings(self, allowed, words):
        """
        :type allowed: str
        :type words: List[str]
        :rtype: int
        """
        ans = 0
        for word in words:
            allowed_str = True
            for ch in word:
                if ch not in allowed:
                    allowed_str = False
                    continue
            if allowed_str:
                ans+=1
            
        return ans;
        

File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.cpp
bool isValidString(vector<int> &v,string word){
            for(auto ch:word){
                if(v[ch-'a']==0) return false;
            }
    return true;
}

class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        int ans=0;
        vector<int> v(26,0);
        for(auto it:allowed){
            v[it-'a']=1;
        }
        for (auto word:words){
           if(isValidString(v,word)) ans++; 
        }
        return ans;
    }
};

